docs/Design/Null Checking.md

```diff
@@ -0,0 +1,148 @@
+# Null Checking
+
+Creates checks for if the passed variable has value null. 
+

```

Comment:
You don't need a blank line between lines in the same paragraph. See other design docs for standards.

*(I use the phrases "blank line" and "end line" differently - the former indicates a full line between them, and end line just means they're on subsequent lines)*